src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +10,31 @@ public class ReadWriteList<E> {
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        try {
+            lock.writeLock().lock();
+            lock.readLock().lock();
```

Comment:
Almost correct. Basically, the idea is that for read operations you should use readLock, for write operations writeLock. When threads lock on readLock, they are not blocked, till some thread doesn't lock on writeLock, then all  who want to lock on readLock should wait while that one thread holds the writeLock. So here you can remove locking on readLock